style_guide.md

```diff
@@ -0,0 +1,270 @@
+# Yield Style Guide
+
+This guide is intended to provide coding conventions for writing solidity code. This guide should be thought of as an evolving document that will change over time as useful conventions are found and old conventions are rendered obsolete.
+
+## Code Layout
+1. Pragma statement
+2. Import statements
+3. Interfaces
+4. Libraries
+5. Contract
+
+#### Code Order
+Inside each contract, library or interface, use the following order:
+
+1. Library declarations (using statements)
+2. Constant variables
+3. State variables
+4. Type declarations
+5. Events
+6. Modifiers
+7. Functions
+
+Functions and declarations should be grouped according to their visibility. The exception to this is that an alternative grouping is sensible to reflect context, i.e. functions are related in some manner.
+
+#### Multiple contracts in the same file
+* Each smart contract should be in its own file.
+
+#### Import Statements
+* Import statements must always be placed at the top of the file.
+* They should be explicitly named import statements, not file level imports.
+```solidity
+Import { Something } from “./MySolidityFile.sol”;
+```
+
+## Naming Conventions
+Keep names as short as possible, wihtout losing precision or clarity in communicating its intent.[^1]
```

Comment:
theres been discussion around this in slack recently

Reply:
yup, I placed the most recent discussion btw Alberto and you under the Local and State Variable Names section (line 57-63)

Reply:
Also, what do you think about the practice as laid out in **Mark untrusted contracts**[¶](https://consensys.github.io/smart-contract-best-practices/development-recommendations/general/external-calls/#mark-untrusted-contracts)